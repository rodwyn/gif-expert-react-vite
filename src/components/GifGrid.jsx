import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {
  const {images, isLoading} = useFetchGifs(category);

  return (
    <>
        <h1>{category}</h1>
        {
          isLoading && <h2>Cargando...</h2>
        }
        
        <div className='card-grid'>
          {
            images.map((image) => (
              <GifItem key={image.id} {...image}  />
            ))
          }
        </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};
