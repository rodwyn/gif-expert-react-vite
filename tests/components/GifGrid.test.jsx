import { render, screen } from "@testing-library/react";
import { GifGrid } from "../../src/components/GifGrid";
import { useFetchGifs } from "../../src/hooks/useFetchGifs";

jest.mock('../../src/hooks/useFetchGifs')

describe('Testing <GifGrid /> component.', () => {
  const category = 'pikachu';
  const gifs = [
    {
      id: 'abc',
      title: 'pikachu',
      url: 'https://localhost/pikachu.jpg'
    },
    {
      id: 'bcd',
      title: 'snorlax',
      url: 'https://localhost/snorlax.jpg'
    }
  ];

  test('should render correctly after load.', () => {
    useFetchGifs.mockReturnValue({
      images: gifs,
      isLoading: false
    });

    const {container} = render(<GifGrid category={category} />);

    expect(container).toMatchSnapshot();
  })

  test('should render loading on init.', () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true
    });

    render(<GifGrid category={category} />);

    expect( screen.getByText('Cargando...')).toBeTruthy();
  });

  test('should render images after load', () => {
    useFetchGifs.mockReturnValue({
      images: gifs,
      isLoading: false
    });

    render(<GifGrid category={category} />);

    expect(screen.getAllByRole('img').length).toBe(gifs.length)

  });
});