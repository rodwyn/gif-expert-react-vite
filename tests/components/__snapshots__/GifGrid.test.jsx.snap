// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGrid /> component. should render correctly after load. 1`] = `
<div>
  <h1>
    pikachu
  </h1>
  <div
    class="card-grid"
  >
    <div
      class="card"
    >
      <img
        alt="pikachu"
        src="https://localhost/pikachu.jpg"
      />
      <p>
        pikachu
      </p>
    </div>
    <div
      class="card"
    >
      <img
        alt="snorlax"
        src="https://localhost/snorlax.jpg"
      />
      <p>
        snorlax
      </p>
    </div>
  </div>
</div>
`;
